src/libraries/System.Private.Xml/src/System/Xml/XPath/Internal/LogicalExpr.cs

```diff
@@ -71,18 +72,18 @@ public override object Evaluate(XPathNodeIterator nodeIterator)
         //                  Number,                       String,                        Boolean,                     NodeSet,                      Navigator
         private static readonly cmpXslt[][] s_CompXsltE =
```

Comment:
should this be `cmpXslt?[][]` instead of `null!` all over?

Reply:
Any idea on how we can Assert that we are always aiming to a not-null element?

Reply:
depending on number of usages of this array inside the code I'd either add Debug.Assert per usage or leave this as is (it's an implementation detail in the end so I don't feel strongly on changing this)